camel/models/nvidia_model.py

```diff
@@ -0,0 +1,155 @@
+from camel.types import ModelType
+from camel.utils import BaseTokenCounter, OpenAITokenCounter, api_keys_required
+
+
+class NvidiaModel(BaseModelBackend):
+    r"""NVIDIA API in a unified BaseModelBackend interface.
+
+    Args:
+        model_type (Union[ModelType, str]): Model for which a backend is
+            created, one of NVIDIA series.
+        model_config_dict (Optional[Dict[str, Any]], optional): A dictionary
+            that will be fed into:obj:`openai.ChatCompletion.create()`. If
+            :obj:`None`, :obj:`NvidiaConfig().as_dict()` will be used.
+            (default: :obj:`None`)
+        api_key (Optional[str], optional): The API key for authenticating with
+            the NVIDIA service. (default: :obj:`None`)
+        url (Optional[str], optional): The url to the NVIDIA service.
+            (default: :obj:`None`)
+        token_counter (Optional[BaseTokenCounter], optional): Token counter to
+            use for the model. If not provided, :obj:`OpenAITokenCounter(
+            ModelType.GPT_4)` will be used.
+            (default: :obj:`None`)
+    """
+
+    def __init__(
+        self,
+        model_type: Union[ModelType, str],
+        model_config_dict: Optional[Dict[str, Any]] = None,
+        api_key: Optional[str] = None,
+        url: Optional[str] = None,
+        token_counter: Optional[BaseTokenCounter] = None,
+    ) -> None:
+        if model_config_dict is None:
+            model_config_dict = NvidiaConfig().as_dict()
+        api_key = api_key or os.environ.get("NVIDIA_API_KEY")
+        url = url or os.environ.get(
+            "NVIDIA_API_BASE_URL", "https://integrate.api.nvidia.com/v1"
+        )
+        super().__init__(
+            model_type, model_config_dict, api_key, url, token_counter
+        )
+        self._client = OpenAI(
+            timeout=60,
+            max_retries=3,
+            api_key=self._api_key,
+            base_url=self._url,
+        )
+
+    @api_keys_required("NVIDIA_API_KEY")
+    def run(
+        self,
+        messages: List[OpenAIMessage],
+    ) -> Union[ChatCompletion, Stream[ChatCompletionChunk]]:
+        r"""Runs inference of NVIDIA chat completion.
+
+        Args:
+            messages (List[OpenAIMessage]): Message list with the chat history
+                in OpenAI API format.
+
+        Returns:
+            Union[ChatCompletion, Stream[ChatCompletionChunk]]:
+                `ChatCompletion` in the non-stream mode, or
+                `Stream[ChatCompletionChunk]` in the stream mode.
+        """
+        # Define role mapping
+        ROLE_MAP = {"system": "context"}
+
+        # Convert messages in one comprehension with proper typing
+
+        def convert_message(msg: Dict[str, Any]) -> ChatCompletionMessageParam:
+            role = ROLE_MAP.get(msg["role"], msg["role"])
+            content = msg["content"]
+            # type: ignore[typeddict-item]
+            return {"role": role, "content": content}
```

Comment:
could you help me understand why we need to do the convert here?

Reply:
Hi Wendong, thanks for your reviewing. We do the convert here because the Nvidia's API expects system messages to have a role of "context" instead of "system".

Reply:
Personally, I prefer handling it outside of the run.

Reply:
Hi Shengsong and Wendong, I just updated the structure and deleted this part.